packages/sdk-react-native/src/contexts/TurnkeyContext.tsx

```diff
@@ -83,414 +100,350 @@ export const TurnkeyProvider: FC<{
-   *
-   * - Persists the updated session to secure storage.
-   * - Updates the in-memory session state.
-   * - Reschedules the session expiration as a side effect.
-   *
-   * @param updatedSession The new session object.
-   * @throws If updating the session fails.
-   */
-  const updateSession = async (updatedSession: Session) => {
-    try {
-      await Keychain.setGenericPassword(
-        TURNKEY_SESSION_STORAGE,
-        JSON.stringify(updatedSession),
-        {
-          accessible: Keychain.ACCESSIBLE.WHEN_UNLOCKED_THIS_DEVICE_ONLY,
-          service: TURNKEY_SESSION_STORAGE,
-        },
+    if (timeUntilExpiry <= 0) {
+      await expireSession();
+    } else {
+      // schedule expiration
+      expiryTimeoutsRef.current[sessionId] = setTimeout(
+        expireSession,
+        timeUntilExpiry,
       );
-
-      scheduleSessionExpiration(updatedSession.expiry);
-    } catch (error) {
-      throw new TurnkeyReactNativeError("Could not update the session.", error);
     }
   };
 
   /**
+   * Sets the selected session and updates the client instance.
    *
-   * Updates the user's email and/or phone number.
+   * - Retrieves the session associated with the given `sessionId`.
+   * - If the session is valid, initializes a new `TurnkeyClient` and updates the state.
+   * - Saves the selected session id and triggers `onSessionSelected` if provided.
+   * - If the session is expired or invalid, clears the session and triggers `onSessionExpired`.
    *
-   * @param userDetails - Object containing the new email and/or phone number.
-   * @returns The activity response from the user update.
-   * @throws If the client or user is not initialized.
+   * @param sessionId - The id of the session to set as selected.
+   * @returns The selected session if valid, otherwise `undefined`.
    */
-  const updateUser = async (userDetails: {
-    email?: string;
-    phone?: string;
-  }) => {
-    if (client == null || session?.user == null) {
-      throw new TurnkeyReactNativeError("Client or user not initialized");
-    }
-    const parameters: {
-      userId: string;
-      userTagIds: string[];
-      userPhoneNumber?: string;
-      userEmail?: string;
-    } = {
-      userId: session.user.id,
-      userTagIds: [],
-    };
-
-    if (userDetails.phone && userDetails.phone.trim()) {
-      parameters.userPhoneNumber = userDetails.phone;
-    }
-
-    if (userDetails.email && userDetails.email.trim()) {
-      parameters.userEmail = userDetails.email;
-    }
+  const setSelectedSession = useCallback(
+    async ({ sessionId }: { sessionId: string }) => {
+      const session = await getSession(sessionId);
 
-    const result = await client.updateUser({
-      type: "ACTIVITY_TYPE_UPDATE_USER",
-      timestampMs: Date.now().toString(),
-      organizationId: session.user.organizationId,
-      parameters,
-    });
+      if (session?.expiry && session.expiry > Date.now()) {
```

Comment:
very smol nit: can pull this out into a helper like `isValidSession` or `isLiveSession`